From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AJMFactsheets <AJMFactsheets@gmail.com>
Date: Fri, 17 Jan 2020 17:17:54 -0600
Subject: [PATCH] Fix items not falling correctly

Since 1.14, Mojang has added an optimization which skips checking if
an item should fall every fourth tick.

However, Spigot's entity activation range class also has an
optimization which skips ticking active entities every fourth tick.
This can result in a state where an item will never properly fall
due to its move method never being called.

This patch resolves the conflict by offsetting checking an item's
move method from Spigot's entity activation range check.

diff --git a/src/main/java/net/minecraft/server/EntityItem.java b/src/main/java/net/minecraft/server/EntityItem.java
index cc23e1fb3e8519e99bad85a77c945596f4123451..e1372620fcb153f0ff9015b3c948c5307a584418 100644
--- a/src/main/java/net/minecraft/server/EntityItem.java
+++ b/src/main/java/net/minecraft/server/EntityItem.java
@@ -89,7 +89,7 @@ public class EntityItem extends Entity {
                 }
             }
 
-            if (!this.onGround || c(this.getMot()) > 9.999999747378752E-6D || (this.ticksLived + this.getId()) % 4 == 0) {
+            if (!this.onGround || c(this.getMot()) > 9.999999747378752E-6D || this.ticksLived % 4 == 0) { // Paper - Ensure checking item movement is always offset from Spigot's entity activation range check
                 this.move(EnumMoveType.SELF, this.getMot());
                 float f1 = 0.98F;
 
